client/components/data/query-country-states/index.jsx

```diff
@@ -13,13 +13,13 @@ import { isCountryStatesFetching } from 'calypso/state/country-states/selectors'
 import { requestCountryStates } from 'calypso/state/country-states/actions';
 
 class QueryCountryStates extends Component {
-	UNSAFE_componentWillMount() {
+	componentDidMount() {
 		this.request( this.props );
 	}
 
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.countryCode !== nextProps.countryCode ) {
-			this.request( nextProps );
+	componentDidUpdate( prevProps ) {
+		if ( this.props.countryCode !== prevProps.countryCode ) {
+			this.request( prevProps );
```

Comment:
You want to request using the new props, i.e., `this.request( this.props )`.

Or, even better, call just `this.request()` and access `this.props` inside the `request` method.

Reply:
Oops, yes! You're right. Will add a fix 👍🏻

Reply:
Fixed in the original commit and refactored to a function component in https://github.com/Automattic/wp-calypso/pull/52779/commits/f053b63afd3da4f45c4dcbe465b4bfc44932898a